src/command/defaults/ClearCommand.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\command\defaults;
+
+use InvalidArgumentException;
+use pocketmine\command\Command;
+use pocketmine\command\CommandSender;
+use pocketmine\command\utils\InvalidCommandSyntaxException;
+use pocketmine\item\LegacyStringToItemParser;
+use pocketmine\lang\TranslationContainer;
+use pocketmine\player\Player;
+use pocketmine\utils\TextFormat;
+use function array_merge;
+use function count;
+use function is_numeric;
+
+class ClearCommand extends VanillaCommand{
+
+	public function __construct(string $name){
+		parent::__construct(
+			$name,
+			"%pocketmine.command.clear.description",
+			"%pocketmine.command.clear.usage"
+		);
+		$this->setPermission("pocketmine.command.clear.self;pocketmine.command.clear.other");
+	}
+
+	public function execute(CommandSender $sender, string $commandLabel, array $args){
+		if(!$this->testPermission($sender)){
+			return true;
+		}
+
+		if(count($args) > 3){
+			throw new InvalidCommandSyntaxException();
+		}
+
+		$target = null;
+		if($sender instanceof Player){
```

Comment:
It doesn't make sense to check this before checking if args[0] is set. A more intuitive logic flow would be
```
if(args[0] is set){
    //try to find player
}elseif(i am a player){
    //use current player as target
}else{
    //invalid syntax
}
```